### PR TITLE
11804: added threshold in ui for bulk media download

### PR DESCRIPTION
**Pull request by jpdimond**

### CODE REVIEW

File: app/assets/javascripts/views/export_csv_view.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    }
+  }
+
+  spaceLeft() {
+    $.ajax({
+      url: ELMO.app.url_builder.build("media-size"),
+      method: "get",
+      data: "",
+      success: (data) => {
+        $(".calculating-info").hide();
```
Comment (cooperka) [inline app/assets/javascripts/views/export_csv_view.js:34]: For readability and reliability, I like to delegate "ownership" of specific DOM elements to specific methods -- e.g. the `.calculating-info` could always be hidden/shown in the `spaceLeft` method so the logic is self-contained. That way you know that both the `calculateMediaSize` AND `spaceLeft` method are "pure" and don't leave DOM state half-way modified with the assumption that some other method will finish modifying it later (easy for bugs to creep in that way).

So my specific suggestion here would be to move `$(".calculating-info").show();` into this method above `ajax`, and do similar tweaks for other elements if reasonable. It doesn't need to be perfect.

File: app/assets/javascripts/views/export_csv_view.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        $(".media-info").show();
+
+        if (data.space_on_disk) {
+          $("input[type=submit]").removeAttr("disabled");
+
+        } else {
+          $(".error-info").show();
+        }
+        return data;
+      }
```
Comment (cooperka) [inline app/assets/javascripts/views/export_csv_view.js:45]: We probably also want to handle the ajax `error` state here, and show some similar error message like "could not complete your request, try again".

**Review by cooperka — COMMENTED** [review]
Looks great other than minor refactoring in JS, and those feature specs! Thanks for the demo.

File: app/assets/javascripts/views/export_csv_view.js (hunk 1 above)
Comment (jpdimond) [reply app/assets/javascripts/views/export_csv_view.js:36]: thanks for this Kevin! super helpful to learn.

File: app/assets/javascripts/views/export_csv_view.js (hunk 2 above)
Comment (jpdimond) [reply app/assets/javascripts/views/export_csv_view.js:50]: done

File: app/assets/stylesheets/local/responses/_export_csv_modal.scss [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#export-csv-modal {
+  .error_box,
```
[the remark this answers is not in the document]
Comment (jpdimond) [reply app/assets/stylesheets/local/responses/_export_csv_modal.scss:2]: error_box already exists in the code, so i don't want to change...

File: app/assets/javascripts/views/export_csv_view.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      $("input[type=submit]").removeAttr("disabled");
+      $(".media-info").hide();
+      $(".error-info").hide();
+    }
+  }
+
+  spaceLeft() {
+    $(".calculating-info").show();
+
+    $.ajax({
```
Comment (cooperka) [inline app/assets/javascripts/views/export_csv_view.js]: In JS, requests like this happen asynchronously by default, so you should use `async/await` if you want to make sure the call finishes before continuing code execution.

I'm not sure if you've used this much in the past, so with a deadline to deploy this week I went ahead and added it -- you can see in https://github.com/thecartercenter/nemo/pull/850/commits/9fb1f0fa89c7f11ac1037418ba2118ad93d035d0, lemme know if you wanna talk more!

**Review by cooperka — APPROVED** [review]
It works great 👏 I made small tweaks as mentioned in comments.

Waiting for specs to pass, then I will merge and deploy it this week!

File: app/models/utils/bulk_media_packager.rb [hunk 5]
```diff
@@ -25,15 +25,21 @@ def media_objects_scope
...
+    def media_meta
+      {space_on_disk: space_on_disk?, media_size: bytes_to_mb(media_size)}
+    end
+
+    def media_size
+      @media_size ||= media_objects_scope.sum("active_storage_blobs.byte_size")
     end
 
     def space_on_disk?
+      return false if Rails.env.test? && ENV.key?("STUB_DISK_FULL")
```
[the remark this answers is not in the document]
Comment (jpdimond) [reply app/models/utils/bulk_media_packager.rb]: I paired with Tom on this one morning, and the ENV way was the way he recommended. Even tho i tried your way first lol (it wasn't working)

**Pull request merged**